packages/pages/src/common/src/module/internal/getModuleFilepaths.ts

```diff
@@ -0,0 +1,89 @@
+import { globSync } from "glob";
+import path from "path";
+import { Path } from "../../project/path.js";
+import { convertToPosixPath } from "../../template/paths.js";
+import { ProjectStructure } from "../../project/structure.js";
+import { readdirSync } from "fs";
+import { getReactVersion } from "../../template/internal/util.js";
+import { ModuleClientServerRenderTemplates } from "../types.js";
+import { fileURLToPath } from "url";
+
+/**
+ * Get all the module files in the provided module folder path(s).
+ *
+ * If there are files that share the same name between the provided
+ * module folder paths, only the file found in the first visited path
+ * from the list is included.
+ *
+ * @param paths a list of paths to collect module files from
+ * @returns a list of module filepaths
+ */
+export const getModuleFilepaths = (paths: Path[]): string[] => {
+  const moduleFilePaths: string[] = [];
+  const addedFilenames: Set<string> = new Set();
+  paths.forEach((p) => {
+    readdirSync(p.getAbsolutePath(), { withFileTypes: true })
+      .filter((dir) => dir.isDirectory())
+      .map((dir) => dir.name)
+      .forEach((dir) => {
```

Comment:
I don't think we need to do this validation here. It can happen at request time if a file doesn't exist when we try to load the module at a path that has no tsx file.

Reply:
As in don't check if there's only 1 source file? We already have the number of source files though. We're also looking for the singular source file in each module folder and wouldn't want to return multiple for one folder.

For templates we do some validation in this similar step:
https://github.com/yext/pages/blob/ba4647aa17ae53fa744f81530b2012a618c2fc58/packages/pages/src/common/src/template/internal/getTemplateFilepaths.ts

Reply:
Ok we can leave it.